From 3391676e0e81fc63a2bb2c7b5036c9abcf4997bf Mon Sep 17 00:00:00 2001
From: ngc7331 <ngc7331@outlook.com>
Date: Sun, 19 Nov 2023 19:19:21 +0800
Subject: [PATCH 2/6] feat: add LV_PORT_PRINT_CLICK_POS macro

---
 components/lvgl/port/lv_port_indev.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/components/lvgl/port/lv_port_indev.c b/components/lvgl/port/lv_port_indev.c
index ed02ea3..ebc2a8a 100644
--- a/components/lvgl/port/lv_port_indev.c
+++ b/components/lvgl/port/lv_port_indev.c
@@ -333,7 +333,9 @@ static int touch_read(uint16_t *x, uint16_t *y)
     pos_x = (pos_x << 8) | buf[3];
     uint16_t pos_y = buf[4] & 0x0F;
     pos_y = (pos_y << 8) | buf[5];
+#ifdef LV_PORT_PRINT_CLICK_POS
     printf("pos: (%-3u, %-3u)[%u]\r\n", pos_x, pos_y, status);
+#endif
     if (status != 8) {
         return 0;
     }
-- 
2.34.1

