From 5100aa3bcdfd7e2cddaaf2a53f8b7588275fc24f Mon Sep 17 00:00:00 2001
From: ngc7331 <ngc7331@outlook.com>
Date: Tue, 19 Dec 2023 17:37:02 +0800
Subject: [PATCH 5/6] feat: add warning if buffer is too small

---
 .../c906/m1s_c906_xram/src/m1s_c906_xram_wifi.c       | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/components/sipeed/c906/m1s_c906_xram/src/m1s_c906_xram_wifi.c b/components/sipeed/c906/m1s_c906_xram/src/m1s_c906_xram_wifi.c
index 44db3a2..21450fd 100644
--- a/components/sipeed/c906/m1s_c906_xram/src/m1s_c906_xram_wifi.c
+++ b/components/sipeed/c906/m1s_c906_xram/src/m1s_c906_xram_wifi.c
@@ -110,8 +110,15 @@ int m1s_xram_wifi_http_response(char *buf, int *len) {
         return -1;
     }
 
-    *len = *len < http_response_len ? *len : http_response_len;
-    strncpy(buf, http_response_buf, *len);
+    if (*len <= http_response_len) {
+        printf("m1s_xram_wifi_http_response: buffer too small, %d/%d bytes copied\r\n", *len-1, http_response_len);
+        strncpy(buf, http_response_buf, *len - 1);
+        buf[*len - 1] = '\0';
+    } else { // *len >= http_response_len + 1 (NULL terminated)
+        printf("m1s_xram_wifi_http_response: %d bytes copied\r\n", *len);
+        strncpy(buf, http_response_buf, (*len = http_response_len));
+        buf[http_response_len] = '\0';
+    }
 
     http_response_len = 0;
     vPortFree(http_response_buf);
-- 
2.34.1

